Fix exceptional weither icon size (#6425)

gibwar opened the pull request:
The weather icon for the `exceptional` state uses a different DOM layout
than the other icons, using a `.weather-icon` class that sets the size
to 64px (52px on narrow). This updates the `forecast-icon > *` class to
set the correct variable to match the expected `40px` and works on all
sizes from `veryverynarrow` to normal.

This does remove the sizes for the `.weather-icon` class, but seeing that
this is a self-contained card, I don't think there is any impact outside
this change.


<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
This fixes the exceptional icon size to match the 40px of the other icons.
### Issue
![issue](https://user-images.githubusercontent.com/1493527/90342368-faa62d00-dfc4-11ea-9c8c-baea997cae9a.PNG)

### Fix
![fix](https://user-images.githubusercontent.com/1493527/90342371-009c0e00-dfc5-11ea-9fe2-6936f72f8f13.PNG)
(the above screenshot shows they veryverynarrow, verynarrow, narrow, and normal layouts, in that order)

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
Configure NWS integration from GUI. Note, since this uses live weather data,
I overrode `/src/panels/lovelace/cards/hui-weather-forecast-card.ts` line 257
to read:
```typescript
${getWeatherStateIcon('exceptional', null)
```
to get the screenshots above.

## Additional information

- This PR fixes or closes issue: fixes #6425
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

homeassistant commented:
Hi @gibwar, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#6634).

Once you do that we will be able to review and accept this pull request.

Thanks!

zsarnett commented on src/panels/lovelace/cards/hui-weather-forecast-card.ts:
```diff
@@ -353,10 +353,6 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
           height: 64px;
         }
 
-        .weather-icon {
```
Why was this removed?

zsarnett replied:
What if the exception icon is the main icon and not in the forecast

gibwar replied:
Ah, I see how it can be used up there. Give me a few and I'll push a fix.

gibwar replied:
Fixed.

zsarnett commented on src/panels/lovelace/cards/hui-weather-forecast-card.ts:
```diff
@@ -469,10 +466,6 @@ class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
           width: 52px;
         }
 
-        :host([narrow]) .weather-icon {
```
Same here

gibwar replied:
Fixed by restricting the selector to `:host([narrow]) .icon-image .weather-icon` so it only applies if it's the main icon and not a forecast icon.

gibwar committed:
Fix exceptional weither icon size (#6425)

The weather icon for the `exceptional` state uses a different DOM layout
than the other icons, using a `.weather-icon` class that sets the size
to 64px (52px on narrow). This updates the `forecast-icon > *` class to
set the correct variable to match the expected `40px` and works on all
sizes from `veryverynarrow` to normal.

gibwar commented:
Updated the PR resolving @zsarnett's comments. I missed that `.weather-icon` was also used for the main icon as well as the exceptional icon. Resolved by making sure that the 40px applies to forecast icons and restricted the main image to apply to ` .icon-image`.
